ibis/expr/operations/strings.py

```diff
@@ -63,6 +63,16 @@ class Substring(Value):
     shape = rlz.shape_like("arg")
 
 
+@public
+class StringSlice(Value):
+    arg: Value[dt.String]
+    start: Optional[Value[dt.Integer]] = None
+    end: Optional[Value[dt.Integer]] = None
```

Comment:
a better name would be (start, stop), but python uses (start, end) in slices, so I decided to be consistent with that.